File: src/node/pi.js
exports.pi_opt = (border) => {
  let pi = 0
  let x = 1
  while (x < border) {
    pi += 4 / x
    x += 2
    pi -= 4 / x
    x += 2
  }
  return pi
}

exports.pi = (border) => {
  let pi = 0
  for (let x = 1; x < border; x++) {
    if ((x - 1) % 2 === 0) {
      pi += (x - 1) % 4 === 0 ? 4 / x : -4 / x
    }
  }
  return pi
}

File: crates/neon/src/pi.rs
pub fn pi (border: i32) -> f64 {
  let mut pi:f64 = 0.0;
  for x in 1..border {
    if (x-1)%2 == 0 {
      pi += if (x-1)%4 == 0 { 4.0/(x as f64) } else { -4.0/(x as f64) }; 
    }
  };
  pi
}

pub fn pi_opt (border: i32) -> f64 {
  let mut pi:f64 = 0.0;
  let mut x: f64 = 1.0;
  while (x as i32) < border {
    pi += 4.0 / x;
    x += 2.0;
    pi -= 4.0 / x;
    x +=2.0;
  }
  pi
}
